lib/daru/io/importers/redis.rb

```diff
@@ -0,0 +1,66 @@
+require 'daru/io/importers/linkages/redis'
+require 'json'
+
+module Daru
+  module IO
+    module Importers
+      class Redis
+        def initialize(connection={}, *keys)
+          @client = get_client(connection)
+          @keys   = choose_keys(*keys)
+        end
+
+        def load
+          vals = @keys.map { |key| ::JSON.parse(@client.get(key)) }
+
+          case ::JSON.parse(@client.get(@keys[0]))
+          when Array
+            case ::JSON.parse(@client.get(@keys[0]))[0]
+            when Hash
+              # Array of hashes
+              # key a :
+              # [
+              #   { x: 1, y: 2 },
+              #   { x: 3, y: 4 }
+              # ]
+              Daru::DataFrame.new vals.flatten
+            else
+              # Hash containing Array
+              # key a :
+              # {
+              #   x: [1,2,3,4]
+              #   y: [5,6,7,8]
+              # }
+              Daru::DataFrame.rows vals.transpose, order: @keys
+            end
+          when Hash
+            # Array containing Hash
+            # [
+            #   key a: { x: 1, y: 2 }
+            #   key b: { x: 3, y: 4 }
+            # ]
+            Daru::DataFrame.new vals.flatten, index: @keys
+          end
+        end
+
+        private
+
+        def choose_keys(*keys)
+          if keys.count.zero?
+            @client.keys
```

Comment:
Please, always try to think about real-life cases, not only test cases with 3 keys in Redis. In real life, there could be hundreds. Or millions. `Redis#keys` is extremely ineffective (`O(N)`), and API that allows passing keys one-by-one is not usable for most of imaginable situations. Take a look at [#scan](http://www.rubydoc.info/gems/redis/3.3.1/Redis#scan-instance_method) family of functions. It allows:
* effectively iterate through keys;
* limit amount of fetch (like "let's fetch first 100 values from DB and play with them for debugging, and then run our study for full database");
* provide key patterns (when people use meaningful/namespaced keys instead of monotonous ones).

Looks like an ideal fit for our task.

NB: I am not a Redis expert. I've just thought a bit, then googled for "Redis best practices" and found out `scan` advice [at the very top of very first article](http://objectrocket.com/blog/how-to/10-quick-tips-about-redis/). It would not hurt if you'll follow my example ;)